source/speechDictHandler/__init__.py

```diff
@@ -115,9 +114,27 @@ def sub(self, text):
 				del self[index]
 		return text
 
+
+# useful for #14689 (see below)
+IGNORE_TRAILING_WHITESPACE_LENGTH = 100
+
 def processText(text):
 	if not globalVars.speechDictionaryProcessing:
 		return text
+	# #14689: API level < 2 UIA consoles have many blank lines, which slows processing to a halt
+	focus = globalVars.focusObject

```

Comment:
Why not use api.getReviewPosition?

```suggestion
	textInfo = api.getReviewPosition()
```

Reply:
This should at the very least use `api.getFocusObject`. `globalVars.focusObject` should not be used directly, as per the comment attached.